src/Microsoft.Diagnostics.Monitoring.WebApi/Operation/EgressOperationService.cs

```diff
@@ -46,7 +46,12 @@ private async Task ExecuteEgressOperation(EgressRequest egressRequest, Cancellat
 
                 try
                 {
-                    ExecutionResult<EgressResult> result = await egressRequest.EgressOperation.ExecuteAsync(_serviceProvider, token);
+                    Task<ExecutionResult<EgressResult>> executeTask = egressRequest.EgressOperation.ExecuteAsync(_serviceProvider, token);
+
+                    await egressRequest.EgressOperation.Started.WithCancellation(token);
```

Comment:
nit: Let's use `WaitAsync` instead of `WithCancellation`; the latter is an extension method that we added to allow this cancellation notion in pre-.NET 6 versions and shouldn't be necessary anymore.